src/all/imhentai/src/eu/kanade/tachiyomi/extension/all/imhentai/IMHentai.kt

```diff
@@ -216,45 +187,38 @@ class IMHentai(override val lang: String, private val imhLang: String) : ParsedH
 
     private val json: Json by injectLazy()
 
-    override fun fetchPageList(chapter: SChapter): Observable<List<Page>> {
-        val gifPages = mutableListOf<Int>()
-        return client.newCall(GET("$baseUrl${chapter.url}"))
-            .asObservableSuccess()
-            .concatMap {
-                val document = it.asJsoup()
-                getGifPages(document, gifPages)
-                client.newCall(buildPageListRequest(document))
-                    .asObservableSuccess()
-            }.map { response ->
-                apiPageListParse(response.asJsoup(), gifPages)
-            }
-    }
-
-    private fun getGifPages(document: Document, gifPages: MutableList<Int>) {
-        val imageFormats = document.selectFirst("script:containsData(var g_th)").data()
-            .substringAfter("$.parseJSON('").substringBefore("');").trim()
-        json.parseToJsonElement(imageFormats).jsonObject.forEach { pair ->
-            val isGif = pair.value.jsonPrimitive.content.startsWith("g")
-            if (isGif) gifPages.add(pair.key.toInt())
+    override fun pageListParse(document: Document): List<Page> {
+        val image_dir = document.select("#image_dir").`val`()
+        val gallery_id = document.select("#gallery_id").`val`()
+        val u_id = document.select("#u_id").`val`().toInt()
+
+        val random_server = when (u_id) {
+            in 1..274825 -> "m1.imhentai.xxx"
+            in 274826..403818 -> "m2.imhentai.xxx"
+            in 403819..527143 -> "m3.imhentai.xxx"
+            in 527144..632481 -> "m4.imhentai.xxx"
+            else -> "m5.imhentai.xxx"
         }
-    }
 
-    private fun apiPageListParse(document: Document, gifs: List<Int>): List<Page> {
-        return document.select("a").mapIndexed { i, element ->
-            Page(
-                i,
-                element.attr("href"),
-                if (gifs.any { page -> page == i + 1 }) {
-                    element.select("img.lazy.preloader").attr("data-src").replace("t.jpg", ".gif")
-                } else {
-                    element.select("img.lazy.preloader").attr("data-src").replace("t.", ".")
-                }
-            )
+        val images = json.parseToJsonElement(
+            document.selectFirst("script:containsData(var g_th)").data()
+                .substringAfter("$.parseJSON('").substringBefore("');").trim()
+        ).jsonObject
+        val pages = mutableListOf<Page>()
+
+        for (image in images) {
+            val iext = image.value.toString().replace("\"", "").split(",")[0]
+            val iext_pr = when (iext) {
+                "p" -> "png"
+                "b" -> "bpm"
```

Comment:
I probably mistyped in the code suggestion...

```suggestion
                "b" -> "bmp"
```